### PR TITLE
Gunrobot

### PR DESCRIPTION
**Pull request by michbr**

### CODE REVIEW

File: Assets/Scripts/Attributes/Endeavour/ShootAction.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    public ShootAction(EndeavourFactory factory, RobotController controller, List<Goal> goals, Dictionary<TagEnum, Tag> tags)
+        : base(factory, controller, goals, tags) {
+        target = getTagOfType<Tag>(TagEnum.Player);
+        name = "shoot";
+    }
+
+    public override bool isStale() {
+        bool stale = !getController().knowsTarget(target.getLabelHandle());
+        return stale;
```
Comment (semimono) [inline Assets/Scripts/Attributes/Endeavour/ShootAction.cs]: Reduce this to one line?

File: Assets/Scripts/GenericRifle.cs [hunk 2]
```diff
@@ -0,0 +1,232 @@
+﻿using UnityEngine;
+using UnityEngine.Networking;
+
+public class GenericRifle : NetworkBehaviour {
```
Comment (semimono) [inline Assets/Scripts/GenericRifle.cs]: We should make the player's rifle scripts inherit from this one as soon as possible.

File: Assets/Prefabs/GameControl/SpawnBase.prefab [hunk 3]
```diff
@@ -308,6 +308,7 @@ MonoBehaviour:
   - {fileID: 128374, guid: 513924346bbb5814c8b8b5ea37c37576, type: 2}
   - {fileID: 1000012390531196, guid: 4058d0703e17c9c4fa0723e7936363ed, type: 2}
   - {fileID: 1482100421143698, guid: 41101aa4c0e18c349857c197c7fc2cfd, type: 2}
+  - {fileID: 1098164653674198, guid: 9b26d4c1ee8ddf04fb53470fb41d0709, type: 2}
```
Comment (michbr) [inline Assets/Prefabs/GameControl/SpawnBase.prefab]: I never added the robo rifle. I don't know what this was. I removed it

File: Assets/Scripts/Attributes/Endeavour/Endeavour.cs [hunk 4]
```diff
@@ -26,6 +26,12 @@ public abstract class Endeavour : Prioritizable {
     private int lastFrameEvaluated = -1;
 	private Dictionary<TagEnum, Tag> tagMap;
 
+    private AbstractArms myArms;
+    private HoverJet myJet;
+    private RoboRifle myRifle;
+    private TowerSpawner myTowerSpawner;
```
Comment (michbr) [inline Assets/Scripts/Attributes/Endeavour/Endeavour.cs]: Smart......Let's plan on that

File: Assets/Prefabs/GameControl/SpawnBase.prefab (hunk 3 above)
Comment (semimono) [inline Assets/Prefabs/GameControl/SpawnBase.prefab]: Cool

File: Assets/Scripts/GenericRifle.cs (hunk 2 above)
Comment (michbr) [inline Assets/Scripts/GenericRifle.cs]: Agreed.

**Pull request merged**